File: combine.py
import os

base_path = r'c:\Users\mikel\Desktop\Sele Matematicas'
data_path = os.path.join(base_path, 'data.js')

with open(data_path, 'r', encoding='utf-8') as f:
    data_content = f.read()

years = ['2021', '2022', '2023', '2024']
new_content = ""

for year in years:
    scratch_path = os.path.join(base_path, f'scratch_{year}.js')
    if os.path.exists(scratch_path):
        with open(scratch_path, 'r', encoding='utf-8') as sf:
            new_content += sf.read() + '\n\n'

placeholder = """// -----------------------------------------------------------------------------------------
// NOTA IMPORTANTE PARA EL USUARIO:
// Por favor, PEGA AQUÍ los bloques de código JavaScript que fuimos generando en el chat
// para las convocatorias de 2021, 2022, 2023 y 2024.
// -----------------------------------------------------------------------------------------"""

new_data_content = data_content.replace(placeholder, new_content)

with open(data_path, 'w', encoding='utf-8') as f:
    f.write(new_data_content)

print("Successfully injected 2021-2024 data into data.js")

File: combine.js
const fs = require('fs');
const path = require('path');

const basePath = 'c:/Users/mikel/Desktop/Sele Matematicas';

const dataPath = path.join(basePath, 'data.js');
let dataContent = fs.readFileSync(dataPath, 'utf8');

const years = ['2021', '2022', '2023', '2024'];
let newContent = '';

for (const year of years) {
  const p = path.join(basePath, `scratch_${year}.js`);
  if (fs.existsSync(p)) {
    newContent += fs.readFileSync(p, 'utf8') + '\n\n';
  }
}

// Replace the placeholder comment with the new content
const placeholder = `// -----------------------------------------------------------------------------------------
// NOTA IMPORTANTE PARA EL USUARIO:
// Por favor, PEGA AQUÍ los bloques de código JavaScript que fuimos generando en el chat
// para las convocatorias de 2021, 2022, 2023 y 2024.
// -----------------------------------------------------------------------------------------`;

dataContent = dataContent.replace(placeholder, newContent);

fs.writeFileSync(dataPath, dataContent, 'utf8');
console.log('Successfully injected 2021-2024 data into data.js');
